### ISSUE TITLE
Exit code always 1 with type check errors and `--force` active

### ISSUE THREAD

**Issue by lepus2589** [palantir/tslint#2087]
### Bug Report

- __TSLint version__: `4.3.1`
- __TypeScript version__: `2.1.5`
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
(x: Number): void => {
    // x is not used in here
}
```

with `tslint.json` configuration: irrelevant

#### Actual behavior

Type checking is activated via the `--type-check` option as well as the `--force` switch. `noUnusedParameters` is active in the `tsconfig.json`, which is given to TSLint with the `--project` argument. The Typescript compiler will of course flag the parameter of the above function as unused.
The file [runner.js](https://github.com/palantir/tslint/blob/master/src/runner.ts#L174) prints the error message (`'x' is declared but never used.`) using an Error. It is not caught anywhere, so the exit code is 1 despite `--force` being used and the (valid) error message is followed by a stack trace. 
Is this the intended behaviour?

#### Expected behavior

I would expect TSLint to print the error messages emitted by the Typescript compiler as linting errors, without an ugly stack trace containing TSLint source files. If not that, then at least when using the `--force` flag, the exit code should be 0, as stated in the documentation.

**Comment by calidion** [comment]
any progress on this?
tslint with errors but still returns 1

**Comment by adidahiya** [comment]
@calidion the issue is marked as closed & fixed. if you're still seeing it you should probably open a new bug report. note that the `--type-check` option is deprecated.
